src/bin/console/console.cpp

```diff
@@ -215,36 +215,55 @@ int Console::_eval_command(const CommandFunction& func, const std::string& comma
   return static_cast<int>(func(args));
 }
 
-int Console::_eval_sql(const std::string& sql) {
-  if (_explicitly_created_transaction_context != nullptr) {
-    _sql_pipeline = std::make_unique<SQLPipeline>(sql, _explicitly_created_transaction_context);
-  } else {
-    _sql_pipeline = std::make_unique<SQLPipeline>(sql);
-  }
-
+bool Console::_initialize_pipelines(const std::string& sql) {
   try {
-    _sql_pipeline->get_result_table();
+    if (_explicitly_created_transaction_context != nullptr) {
+      _sql_pipelines = SQLPipeline::from_sql_string(sql, _explicitly_created_transaction_context);
+    } else {
+      _sql_pipelines = SQLPipeline::from_sql_string(sql);
+    }
   } catch (const std::exception& exception) {
-    out(std::string(exception.what()) + "\n");
-    if (_sql_pipeline->transaction_context() && _sql_pipeline->transaction_context()->aborted()) {
-      out("The transaction has been rolled back.\n");
+    out(std::string(exception.what()) + '\n');
+    return false;
+  }
+
+  return true;
+}
+
+int Console::_eval_sql(const std::string& sql) {
+  if (!_initialize_pipelines(sql)) return ReturnCode::Error;
+
+  for (auto& pipeline : _sql_pipelines) {
+    try {
+      pipeline.get_result_table();
+    } catch (const std::exception& exception) {
+      out(std::string(exception.what()) + "\n");
+      if (pipeline.transaction_context() && pipeline.transaction_context()->aborted()) {
+        out("The transaction has been rolled back.\nAll previous statements have been commited.\n");
```

Comment:
Mmmh? If we have an explicitly created tx context, it should be *completely* rolled back. That is the point of a rollback.